### PR TITLE
Add imperial unit support

### PR DESCRIPTION
**Pull request by Rocketct**
Added imperial units support to MKRENV library

- Added enum to units management 
- Renamed  readLux in readIlluminance to comply the others illuminance scale factors
- Added scale factor support to pressure temperature and illuminance API
- added example to show how to read in Fahrenheit and foot candle the temperature and the illuminance sensor

cc/ @sandeepmistry

### CODE REVIEW

File: examples/ReadSensorsImperial/ReadSensorsImperial.ino [hunk 1]
```diff
@@ -0,0 +1,68 @@
+/*
+  MKR ENV Shield - Read Sensors Imperial
+
+  This example reads the sensors on-board the MKR ENV shield
+  and prints them in imeprial units to the Serial Monitor once a second.
```
Comment (per1234) [inline examples/ReadSensorsImperial/ReadSensorsImperial.ino]:
```suggestion
  and prints them in imperial units to the Serial Monitor once a second.
```

File: examples/ReadSensorsImperial/ReadSensorsImperial.ino [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  while (!Serial);
+
+  if (!ENV.begin()) {
+    Serial.println("Failed to initialize MKR ENV shield!");
+    while (1);
+  }
+}
+
+void loop() {
+  // Seting to FAHRENHEIT and FOOTCANDLE readTemperature and readIlluminance
```
Comment (per1234) [inline examples/ReadSensorsImperial/ReadSensorsImperial.ino]:
```suggestion
  // Setting readTemperature to FAHRENHEIT and readIlluminance to FOOTCANDLE
```

File: examples/ReadSensorsImperial/ReadSensorsImperial.ino [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  // allow to reads in imperial units the temperature and illuminance sensor
```
Comment (per1234) [inline examples/ReadSensorsImperial/ReadSensorsImperial.ino]:
```suggestion
  // allows reading the temperature and illuminance sensors in imperial units
```

**Review by per1234 — CHANGES_REQUESTED** [review]

File: examples/ReadSensorsImperial/ReadSensorsImperial.ino (hunk 2 above)
Comment (sandeepmistry) [reply examples/ReadSensorsImperial/ReadSensorsImperial.ino]: I think this should be reworded to passing in `FAHRENHEIT` as the units argument ...

File: examples/ReadSensorsImperial/ReadSensorsImperial.ino [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  Serial.print(humidity);
+  Serial.println(" %");
+
+  Serial.print("Pressure    = ");
+  Serial.print(pressure);
+  Serial.println(" kPa");
+
+  Serial.print("Illuminance = ");
+  Serial.print(illuminance);
+  Serial.println(" mc");
```
Comment (sandeepmistry) [inline examples/ReadSensorsImperial/ReadSensorsImperial.ino]: Shouldn't the abbreviation here be `fc`?

https://en.wikipedia.org/wiki/Foot-candle

File: examples/ReadSensorsImperial/ReadSensorsImperial.ino [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    while (1);
+  }
+}
+
+void loop() {
+  // Seting to FAHRENHEIT and FOOTCANDLE readTemperature and readIlluminance
+  // allow to reads in imperial units the temperature and illuminance sensor 
+  float temperature = ENV.readTemperature(FAHRENHEIT);
+  float humidity    = ENV.readHumidity();
+  float pressure    = ENV.readPressure();
```
Comment (sandeepmistry) [inline examples/ReadSensorsImperial/ReadSensorsImperial.ino]: I think we are also missing the imperial units for pressure? Is it PSI?

Comment (sandeepmistry) [reply examples/ReadSensorsImperial/ReadSensorsImperial.ino]: Here they use inches? https://www.wunderground.com/weather/us/ny/manhattan/40.75%2C-74.00

File: examples/ReadSensorsImperial/ReadSensorsImperial.ino (hunk 4 above)
Comment (Rocketct) [reply examples/ReadSensorsImperial/ReadSensorsImperial.ino]: done

File: examples/ReadSensorsImperial/ReadSensorsImperial.ino (hunk 1 above)
Comment (Rocketct) [reply examples/ReadSensorsImperial/ReadSensorsImperial.ino]: done

File: examples/ReadSensorsImperial/ReadSensorsImperial.ino (hunk 5 above)
Comment (Rocketct) [reply examples/ReadSensorsImperial/ReadSensorsImperial.ino]: `the pound-force per square inch (psi) is the traditional unit of pressure in the imperial and US customary ` then i add the PSI support

File: examples/ReadSensorsImperial/ReadSensorsImperial.ino (hunk 2 above)
Comment (Rocketct) [reply examples/ReadSensorsImperial/ReadSensorsImperial.ino]: done like as suggested by @sandeepmistry

File: examples/ReadSensorsImperial/ReadSensorsImperial.ino (hunk 5 above)
Comment (Rocketct) [reply examples/ReadSensorsImperial/ReadSensorsImperial.ino]: done

File: examples/ReadSensorsImperial/ReadSensorsImperial.ino (hunk 3 above)
Comment (Rocketct) [reply examples/ReadSensorsImperial/ReadSensorsImperial.ino]: done

**Review by sandeepmistry — APPROVED** [review]

File: examples/ReadSensorsImperial/ReadSensorsImperial.ino [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+  if (!ENV.begin()) {
+    Serial.println("Failed to initialize MKR ENV shield!");
+    while (1);
+  }
+}
+
+void loop() {
+  // Passing in FAHRENHEIT as the unit paramter to ENV.readTemperature(...),
+  // PSI to readPressure(...) and FOOTCANDLE to readIlluminance(...)
+  // allows you to read the sensors values in imperial units
```
Comment (per1234) [inline examples/ReadSensorsImperial/ReadSensorsImperial.ino]:
```suggestion
  // allows you to read the sensor values in imperial units
```

File: examples/ReadSensorsImperial/ReadSensorsImperial.ino [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+  while (!Serial);
+
+  if (!ENV.begin()) {
+    Serial.println("Failed to initialize MKR ENV shield!");
+    while (1);
+  }
+}
+
+void loop() {
+  // Passing in FAHRENHEIT as the unit paramter to ENV.readTemperature(...),
```
Comment (per1234) [inline examples/ReadSensorsImperial/ReadSensorsImperial.ino]:
```suggestion
  // Passing in FAHRENHEIT as the unit parameter to ENV.readTemperature(...),
```

**Review by per1234 — CHANGES_REQUESTED** [review]

**Review by per1234 — APPROVED** [review]

**Pull request merged**